cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go

```diff
@@ -56,3 +85,160 @@ func (s *Server) GetClient(ctx context.Context) (dynamic.Interface, error) {
 	}
 	return client, nil
 }
+
+// GetManager ensures a manager is available and uses it to return the client.
+func (s *Server) GetManager() (assetsvc_utils.AssetManager, error) {
+	if s.manager == nil {
+		return nil, status.Errorf(codes.Internal, "server not configured with manager")
+	}
+	manager := s.manager
+	return manager, nil
+}
+
+// GetAvailablePackageSummaries returns the available packages based on the request.
+func (s *Server) GetAvailablePackageSummaries(ctx context.Context, request *corev1.GetAvailablePackageSummariesRequest) (*corev1.GetAvailablePackageSummariesResponse, error) {
+	contextMsg := ""
+	if request.Context != nil {
+		contextMsg = fmt.Sprintf("(cluster=[%s], namespace=[%s])", request.Context.Cluster, request.Context.Namespace)
+	}
+
+	log.Infof("+helm GetAvailablePackageSummaries %s", contextMsg)
+
+	// Check the request context (namespace and cluster)
+	namespace := ""
+	if request.Context != nil {
+		if request.Context.Cluster != "" {
+			return nil, status.Errorf(codes.Unimplemented, "Not supported yet: request.Context.Cluster: [%v]", request.Context.Cluster)
+		}
+		namespace = request.Context.Namespace
+	}
+	// Check the requested namespace: if any, return "everything a user can read";
+	// otherwise, first check if the user can access the requested ns
+	if namespace == "" {
+		// TODO(agamez): not including a namespace means that it returns everything a user can read
+		return nil, status.Errorf(codes.Unimplemented, "Not supported yet: not including a namespace means that it returns everything a user can read")
+	} else {
```

Comment:
I do agree (I'll change it). However, I decided to leave the `else` statement since, at some point, the `if` branch will have some logic. This way, fewer git changes (aka indentation) when we add the "no namespace" case.
Anyway, happy to remove it.

Reply:
Yep, up to you (I don't mind, just a general note for clarity).